node/src/actors/chain_manager/mod.rs

```diff
@@ -1483,6 +1488,32 @@ impl ChainManager {
             Some(sync_target.superblock.checkpoint * superblock_period)
         }
     }
+
+    /// Let JSON-RPC clients know that the blocks in the previous superblock can now
+    /// be considered consolidated
+    fn notify_previous_superblock_consolidation(&mut self, ctx: &mut Context<ChainManager>) {
+        let superblock_period = u32::from(self.consensus_constants().superblock_period);
+        let final_epoch = self
+            .previous_chain_state
+            .superblock_state
+            .get_beacon()
+            .checkpoint
+            .checked_mul(superblock_period)
+            .expect("Multiplying a superblock ID by `superblock_period` should never overflow");
+        let initial_epoch = final_epoch.saturating_sub(superblock_period);
+        let beacons = self.handle(
+            GetBlocksEpochRange::new_with_limit(initial_epoch..final_epoch, 0),
+            ctx,
+        );
```

Comment:
Oh my... we are storing all block headers in memory! :rofl:

Reply:
It's only a mapping of block_epoch to block_hash, 4 bytes + 32 bytes. That's about 35MB for 1 million blocks, so we can wait a few million blocks until it becomes a problem.